### ISSUE TITLE
Feature request: Linter usage of properties/methods in non object

### ISSUE THREAD

**Issue by nelson6e65** [php-integrator/atom-linter#54]
First, this is an amazing set of packages very useful! :surfer:

It would be useful a linter to warn about usage of object features in an non object. Example:

```php
<?php

// Other class methods . . .
public function doSomething(SomeClass $obj1, $obj2, $obj3 = null)
{
    $obj4 = 'Something';

    $obj1->someMethod(); // No warnings in this meaning here
    $obj2->someProperty = false; // Only notable in runtime, instance type unknow
    $obj3->someProperty = true; // PHP Runtime error: usage on null (preventable by a warning)
    $obj4->getSome(); // PHP runtime error: Usage of method on an non object instance (preventable)
    
    $obj1 = 'I\'m not an object as used to be';
    $obj1->someMethod(); // Runtime error as obj4, also preventable
}
```

At least for class methods and functions.

**Comment by Gert-dev** [comment]
Thanks!

You are absolutely right, in fact you can already enable an experimental version of this via the package settings under `Show unknown members (experimental)`.

It is currently experimental as it may report false positives as it requires rather advanced static type analysis, including things such as conditionals. There are a couple of tickets in the core about rectifying that, such as [1], [2], [3], [4], and other tickets. After these are implemented, I also need to take a look at its performance, as it is a tad too slow to be constantly active.

In short, it's on the radar, but due to its advanced nature will take some time to implement properly. I'll close this as there are already several other tickets related to it in the core. But thanks for taking the time to report it! 

[1] https://github.com/php-integrator/core/issues/36
[2] https://github.com/php-integrator/core/issues/1
[3] https://github.com/php-integrator/core/issues/37
[4] https://github.com/php-integrator/core/issues/38

**Comment by nelson6e65** [comment]
Nice! Thanks for your answer! :smiley:

**Comment by nelson6e65** [comment]
@Gert-dev `Show unknown members` is very useful :muscle:, but it should run in another process (or something) because it locks other integrator-linters. :turtle:
